### PR TITLE
Fs 2002 bulk account

### PR DESCRIPTION
**Pull request by AbuNavsa**
- Amended api.yml to configure bulk-accounts endpoint
- Created BulkAccount component for retrieval of multiple accounts
- Written get_bulk_accounts() method to query database and return multiple rows corresponding to each requested account ID
- Accounts are returned in a nested dictionary, each accounts data has a key of its own Account ID
- FUTURE WORK: Get roles

### CODE REVIEW

File: core/account.py [hunk 1]
```diff
@@ -51,11 +52,49 @@ def get_account(
     try:
         result = db.session.execute(stmnt)
         account = result.scalars().one()
-        account_schema = AccountSchema()
+        account_schema = AccountSchema()
```
Comment (gio-karunakaran) [inline core/account.py:55]: think there might be some weird spacing here at end of line?

File: core/account.py [hunk 2]
```diff
@@ -51,11 +52,49 @@ def get_account(
...
     except sqlalchemy.exc.NoResultFound:
         return {"error": "No matching account found"}, 404
 
+def get_bulk_accounts(
+    account_id: list,
+) -> Dict:
+    """
+    Get multiple accounts corresponding to the given account ids
+    and return account object schema and status code
+    :param account_id: (str) the id to search
```
Comment (gio-karunakaran) [inline core/account.py:66]: if account_id is a list then we can update this docstring param

File: core/account.py [hunk 3]
```diff
@@ -51,11 +52,49 @@ def get_account(
...
+        account_schema = AccountSchema()
+
+        accounts_metadatas = { 
+            str(account_row.id) : account_schema.dump(account_row) 
+            for account_row in result    
+        }
+
+        return accounts_metadatas, 200
+    except sqlalchemy.exc.NoResultFound:
+        return {"error": "No matching account found"}, 404
```
Comment (gio-karunakaran) [inline core/account.py:91]: great stuff 🙌🏾

File: requirements-dev.txt [hunk 4]
```diff
@@ -1,6 +1,6 @@
 #
-# This file is autogenerated by pip-compile with python 3.10
-# To update, run:
+# This file is autogenerated by pip-compile with Python 3.10
+# by the following command:
```
Comment (gio-karunakaran) [inline requirements-dev.txt:3]: did you mean to change this?

File: requirements-dev.txt [hunk 5]
```diff
@@ -63,6 +63,13 @@ clickclick==20.10.2
     # via
     #   -r requirements.txt
     #   connexion
+colorama==0.4.6
+    # via
+    #   -r requirements.txt
+    #   bandit
+    #   build
+    #   click
+    #   pytest
```
Comment (gio-karunakaran) [inline requirements-dev.txt:72]: these requirements here seem a bit strange, did we mean to add these?

File: requirements.txt [hunk 6]
```diff
@@ -1,8 +1,8 @@
 #
-# This file is autogenerated by pip-compile with python 3.10
-# To update, run:
+# This file is autogenerated by pip-compile with Python 3.10
+# by the following command:
 #
-#    pip-compile
+#    pip-compile requirements.in
```
Comment (gio-karunakaran) [inline requirements.txt:5]: same again here? looks like some lines were not meant to be deleted?

File: requirements.txt [hunk 7]
```diff
@@ -34,6 +34,8 @@ click==8.1.3
     #   flask
 clickclick==20.10.2
     # via connexion
+colorama==0.4.6
+    # via click
```
Comment (gio-karunakaran) [inline requirements.txt:38]: what is colorama being used for?

File: tests/test_accounts.py [hunk 8]
```diff
@@ -187,6 +188,66 @@ def test_get_by_unique_columns(self, flask_test_client):
...
+        for id in account_ids:
+            if count < 1:
+                account_id_arg_url += id
+                count += 1
+            else: account_id_arg_url += "&account_id=" + id
+
+        expected_data_within_response(
+            flask_test_client, account_id_arg_url, expected_response_data, 200
+        )
+
```
Comment (gio-karunakaran) [inline tests/test_accounts.py:249]: great job with the tests!

**Review by gio-karunakaran — COMMENTED** [review]
just a few Qs! Great stuff 🙌🏾

File: core/account.py [hunk 9]
```diff
@@ -51,11 +52,49 @@ def get_account(
...
         result = db.session.execute(stmnt)
         account = result.scalars().one()
-        account_schema = AccountSchema()
+        account_schema = AccountSchema()    
         return account_schema.dump(account), 200
     except sqlalchemy.exc.NoResultFound:
         return {"error": "No matching account found"}, 404
 
+def get_bulk_accounts(
+    account_id: list,
```
[the remark this answers is not in the document]
Comment (AbuNavsa) [reply core/account.py:61]: Takes an array of strings in the endpoint so went with list

File: requirements-dev.txt (hunk 5 above)
Comment (AbuNavsa) [reply requirements-dev.txt:72]: Will remove from the changes

File: requirements.txt (hunk 7 above)
Comment (AbuNavsa) [reply requirements.txt:38]: Unsure, was autogenerated but will be removed

File: tests/test_accounts.py (hunk 8 above)
Comment (AbuNavsa) [reply tests/test_accounts.py:249]: Thank you!

File: requirements.txt (hunk 7 above)
Comment (AbuNavsa) [reply requirements.txt]: removed and file matches that of master

File: requirements.txt (hunk 6 above)
Comment (AbuNavsa) [reply requirements.txt]: reverted

File: core/account.py (hunk 2 above)
Comment (AbuNavsa) [reply core/account.py]: Has been amended to describe accurately

**Review by gio-karunakaran — APPROVED** [review]
Nice 1!

**Pull request merged**